### PR TITLE
Code simplification in X509CertInfo

### PR DESCRIPTION
**Pull request by ckelleyRH**
* Simplify equals() with pattern-matching instanceof and removing
unnecessary Boolean literal.
* Combine identical catch clauses
* Use switch expressions and ternary operators where appropriate

### CODE REVIEW

**Review by edewata — APPROVED** [review]
LGTM. I have some comments but feel free to merge.

File: base/src/main/java/org/mozilla/jss/netscape/security/x509/X509CertInfo.java [hunk 1]
```diff
@@ -288,11 +288,7 @@ public byte[] getEncodedInfo(boolean ignoreCache) throws CertificateEncodingExce
      */
     @Override
     public boolean equals(Object other) {
-        if (other instanceof X509CertInfo) {
-            return equals((X509CertInfo) other);
-        } else {
-            return false;
-        }
+        return other instanceof X509CertInfo certInfo && equals(certInfo);
```
[the remark this answers is not in the document]
Comment (ckelleyRH) [reply base/src/main/java/org/mozilla/jss/netscape/security/x509/X509CertInfo.java:291]: Hmm, maybe the parentheses is better. I think the wrapping is a bit jarring for only two elements.

File: base/src/main/java/org/mozilla/jss/netscape/security/x509/X509CertInfo.java [hunk 2]
```diff
@@ -613,78 +607,34 @@ public Object get(String name)
...
+            case ATTR_ALGORITHM -> attrName.getSuffix() == null ? algId : algId.get(attrName.getSuffix());
+            case ATTR_ISSUER -> attrName.getSuffix() == null ? issuer : issuer.get(attrName.getSuffix());
+            case ATTR_VALIDITY -> attrName.getSuffix() == null ? interval : interval.get(attrName.getSuffix());
+            case ATTR_SUBJECT -> attrName.getSuffix() == null ? subject : subject.get(attrName.getSuffix());
+            case ATTR_KEY -> attrName.getSuffix() == null ? pubKey : pubKey.get(attrName.getSuffix());
+            case ATTR_ISSUER_ID -> {
+                if (attrName.getSuffix() == null) {
+                    yield issuerUniqueId;
+                }
+                yield issuerUniqueId == null ? null : issuerUniqueId.get(attrName.getSuffix());
```
[the remark this answers is not in the document]
Comment (ckelleyRH) [reply base/src/main/java/org/mozilla/jss/netscape/security/x509/X509CertInfo.java:622]: Noice, will incorporate!

**Comment by ckelleyRH** [comment]
I incorporated your suggestions, but I used a `boolean` rather than boxed `Boolean` as Sonar was complaining about `null` comparison with boxed Booleans. I'll merge if you're happy with this change!

**Comment by edewata** [comment]
That looks good. Thanks for the update!

**Comment by ckelleyRH** [comment]
Thanks @edewata !

**Pull request merged**